lace_thread_style.py

```diff
@@ -24,111 +25,183 @@
-		"""
-		if verbose: inkex.debug("%s ="%(longColor))
-		longColor = long(longColor)
-		if longColor <0: longColor = long(longColor) & 0xFFFFFFFF
-		hexColor = hex(longColor)[2:-3]
-		hexColor = '#' + hexColor.rjust(6, '0').upper()
-		if verbose: inkex.debug("  %s for color default value"%(hexColor))
-		return hexColor
-
-	def effect(self):
-		"""
-		Effect behaviour.
-		Overrides base class' method and draws something.
-		"""
-		self.options.color = self.getColorString(self.options.color)
-		conversion = self.getUnittouu("1" + self.options.units)
-		if len(self.selected) != 1:
-			inkex.debug('no object selected, or more than one selected')
-			return
-		selected = self.selected.values()[0]
-		if not self.isBezier(selected):
-			inkex.debug('selected element is not a Bezier curve')
-			return
-		self.findCandidatesForStyleChange(selected)
-		self.style = 'fill:none;stroke:{1};stroke-width:{0}'.format(self.options.width*conversion, self.options.color)
-		csp = cubicsuperpath.parsePath(selected.get('d'))
-		self.selected.values()[0].attrib['style'] = self.style
-		self.applyToAdjacent(self.startPoint(csp))
-		self.applyToAdjacent(self.endPoint(csp))
-			
+    """
+    Apply stroke style of selected path to connected paths
+    """
+    def __init__(self):
+        """
+        Constructor.
+        """
+        # Call the base class constructor.
+        inkex.Effect.__init__(self)
+        self.OptionParser.add_option('-t', '--tolerance', action='store', type='float', dest='tolerance', default=0.05, help='tolerance (max. distance between segments)')
+        self.OptionParser.add_option('-u', '--units', action = 'store', type = 'string', dest = 'units', default = 'mm', help = 'The units the measurements are in')
+        self.OptionParser.add_option('-w', '--width', action='store', type='float', dest='width', default='1', help='thread width')
+        self.OptionParser.add_option('-c', '--color', action='store', type='string', dest='color', default='#FF9999', help='thread color')
+
+    def getUnittouu(self, param):
+        " compatibility between inkscape 0.48 and 0.91 "
+        try:
+            return inkex.unittouu(param)
+        except AttributeError:
+            return self.unittouu(param)
+
+    def getColorString(self, longColor, verbose=False):
+        """ Convert the long into a #RRGGBB color value
+            - verbose=true pops up value for us in defaults
+            conversion back is A + B*256^1 + G*256^2 + R*256^3
+        """
+        if verbose: inkex.debug("%s ="%(longColor))
+        longColor = long(longColor)
+        if longColor <0: longColor = long(longColor) & 0xFFFFFFFF
+        hexColor = hex(longColor)[2:-3]
+        hexColor = '#' + hexColor.rjust(6, '0').upper()
+        if verbose: inkex.debug("  %s for color default value"%(hexColor))
+        return hexColor
+
+    def random_color(self):
+        red = random.uniform(0.2, 1)
+        grn = random.uniform(0.2, 1)
+        blu = random.uniform(0.2, 1)

```

Comment:
I've been playing with the values a bit. To analyze how threads run through the lace I think it is better to provide a palette with high contrast colors and repeat them. The color bar at the bottom of Inkscape provides 12 named colors, add black, 40%  and 70% grey or so. Let the user choose how many (or even which ones of the palette) colors to use to match the repeats in the pattern. I'm afraid to pick which ones to use we have to use check boxes and color-names. The daring users might want to tweak these choices in the installed inx file.